cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/ProgramRunDispatcherFactory.java

```diff
@@ -22,18 +22,27 @@
 import io.cdap.cdap.common.conf.Constants;
 import io.cdap.cdap.proto.ProgramType;
 
-import java.util.Objects;
+import java.util.HashSet;
+import java.util.Set;
 
 public class ProgramRunDispatcherFactory {
 
-  private final CConfiguration cConf;
   private final InMemoryProgramRunDispatcher inMemoryProgramRunDispatcher;
+  private final boolean workerPoolEnabled;
+  private final Set<ProgramType> remoteDispatchProgramTypes;
   private RemoteProgramRunDispatcher remoteProgramRunDispatcher;
 
   @Inject
   public ProgramRunDispatcherFactory(CConfiguration cConf, InMemoryProgramRunDispatcher inMemoryProgramRunDispatcher) {
-    this.cConf = cConf;
     this.inMemoryProgramRunDispatcher = inMemoryProgramRunDispatcher;
+    this.workerPoolEnabled = cConf.getBoolean(Constants.SystemWorker.POOL_ENABLE);
+    this.remoteDispatchProgramTypes = new HashSet<>();
+    String[] programTypes = cConf.getStrings(Constants.SystemWorker.DISPATCH_PROGRAM_TYPES);
+    if (programTypes != null) {
+      for (String type : programTypes) {
+        remoteDispatchProgramTypes.add(ProgramType.valueOf(type.toUpperCase()));
```

Comment:
since this is user input, would be good to catch the exception that happens when the type is invalid, log a warning and continue.

Reply:
Done!